screenshot-test/App.screenshot.js

```diff
@@ -0,0 +1,596 @@
+    await page.setViewport(VIEW_PORT_1080);
+    await page.waitFor(LONG_WAITING_TIME);
+    let image = await page.screenshot(getScreenshotConfig('Gene Page with Login'));
+    expect(image).toMatchImageSnapshot({ customSnapshotIdentifier: 'Gene Page with Login' });
+  })
+
+  it('Alteration Page', async() => {
+    await page.goto(`${CLIENT_URL}gene/ABL1/BCR-ABL1%20Fusion`);
+    await page.setViewport(VIEW_PORT_1080);
+    await page.waitFor(WAITING_TIME);
+    let image = await page.screenshot(getScreenshotConfig('Alteration Page with Login'));
+    expect(image).toMatchImageSnapshot({ customSnapshotIdentifier: 'Alteration Page with Login' });
+  })
+
+  it('Account Settings Page', async() =>{
+    await page.goto(`${CLIENT_URL}account/settings`);
+    await page.setViewport(VIEW_PORT_1080);
+    await page.waitFor(WAITING_TIME);
+    let image = await page.screenshot(getScreenshotConfig('Account Settings Page'));
+    expect(image).toMatchImageSnapshot({ customSnapshotIdentifier: 'Account Settings Page' });
+  })
+
+  it('Users Infomation Page', async() =>{
+    await page.goto(`${CLIENT_URL}admin/user-details`);
+    await page.setViewport(VIEW_PORT_1080);
+    await page.waitFor(WAITING_TIME);
+    let image = await page.screenshot(getScreenshotConfig('Users Infomation Page'));
+    expect(image).toMatchImageSnapshot({ customSnapshotIdentifier: 'Users Infomation Page' });
+  })
+
+  it('User Details Page', async() =>{
+    await page.goto(`${CLIENT_URL}users/admin`);
+    await page.setViewport(VIEW_PORT_1080);
+    await page.waitFor(WAITING_TIME);
+    let image = await page.screenshot(getScreenshotConfig('User Details Page'));
+    expect(image).toMatchImageSnapshot({ customSnapshotIdentifier: 'User Details Page' });
+  })
+
+  afterAll(async () => {
+    await browser.close();
+  })
+})
+
+describe('Tests without login', () => {
+  let browser;
+  let page;
+
+  beforeAll(async () => {
+    browser = await puppeteer.launch(browserConfig);
+    page = await browser.newPage();
+    await page.setRequestInterception(true); // Handle UnhandledPromiseRejectionWarning: Error: Request Interception is not enabled! 
+    page.on('request', (request) => {
+      let url = request.url()
+      if (url.includes('recaptcha')){
+        request.abort();
+      }
+      else{
+        if (getRes(url) !== undefined){
+          request.respond(
+            getRes(url)
+          )
+        }
+        else request.continue();
+      }     
+    });
+    await page.goto(`${CLIENT_URL}`);
+    await page.evaluate(() => {
+      localStorage.setItem('localdev', 'true');
+    });
+  })
+
+  it('Home Page', async() => {
+    await page.goto(`${CLIENT_URL}`);
+    await page.setViewport(VIEW_PORT_1080);
+    await page.waitFor(WAITING_TIME);
+    let image = await page.screenshot(getScreenshotConfig('Home Page without Login'));
+    expect(image).toMatchImageSnapshot({ customSnapshotIdentifier: 'Home Page without Login' });
+  })
```

Comment:
I just noticed that these tests designed for with/out login are identical except some string which can be changed through input. I believe you want to design test for both with and without login, correct? Maybe consolidate these tests to a method?

Reply:
I've considered it. But there would be some async errors if wrap these in a method. Maybe we could use it now and I will do more research later.